libcobblersignatures/__init__.py

```diff
@@ -2,3 +2,221 @@
+        """
+        If you want to create an empty file just use this constructor. If you want to import one use this and then run
+        an import function.
+        """
+        self._rootkey = "breeds"
+        self._signaturesjson = {}
+        self._osbreeds = []
+
+    @property
+    def signaturesjson(self) -> dict:
+        """
+        This property represents the json which was im- or exported.
+
+        :raises JSONDecodeError: If invalid json is given
+        :return: The last valid content of the json which the library knows about.
+        """
+        return self._signaturesjson
+
+    @signaturesjson.setter
+    def signaturesjson(self, value: str):
+        """
+        The setter for the exported data structure.
+
+        :param value: The str which will be loaded by the Python json utility.
+        """
+        if value is None:
+            pass
+        self._signaturesjson = json.loads(value)
+
+    @property
+    def osbreeds(self) -> List[OsBreed]:
+        """
+        This property represents the currently manipulated data structures.
+
+        :return: The list with the content which is currently manipulated.
+        """
+        return self._osbreeds
+
+    @osbreeds.setter
+    def osbreeds(self, value: List[OsBreed]):
+        """
+        The setter for the osbreeds property.
+
+        :param value: The list with the OsBreeds.
+        """
+        self._osbreeds = value
+
+    def importsignatures(self, import_type: ImportTypes, source: str):
+        """
+        This is the main import function.
+
+        :param import_type: This is one of the four options from the ImportTypes Enum
+        :param source: A string containing the data to be parsed into a json in the end
+        """
+        # TODO make sure source is never none and reasonable
+        if import_type == ImportTypes.FILE:
+            self._importsignaturesfile(source)
+        elif import_type == ImportTypes.URL:
+            self._importsignaturesurl(source)
+        elif import_type == ImportTypes.STRING:
+            self.signaturesjson = source
+        else:
+            raise ValueError("Please use on of the four given options for the source!")
+
+    def _importsignaturesfile(self, filepath: str):
+        """
+        Internal function to handle the import of a file based import.
+
+        :param filepath: The relative or absolute path. Additionally this may be all path variations which are accepted
+                         by the Python ``open()`` function.
+        """
+        with open(filepath, 'r') as f:
+            self.signaturesjson = f.read()
+
+    def _importsignaturesurl(self, url: str):
+        """
+        Internal function to handle the import of a URL based import. The data behind it should be a well formed JSON.
+
+        :param url: The URL to load the content from. This parameter is handed to urllib.
+        """
+        response = urllib.request.urlopen(url)
+        data = response.read()
```

Comment:
I wonder if this could be a problem on large responses and if reading + decoding the response in chunks is favorable.

Reply:
@agraul and I agreed that while his concerns are more then valid, we will wait for feedback from the users how much we need to safeguard this.